test/unit/devices.test.ts

```diff
@@ -1,378 +1,541 @@
+				{ capability: 'capability-id-2', command: 'command-2', arguments: ['arg1', 'arg2'] },
+			]
+			postSpy.mockResolvedValueOnce(commandResponse)
+
+			expect(await devices.sendCommand(configEntry, cmdList)).toBe(commandResponse)
+			const expectedCommand1 = {
+				component: 'component-id',
+				capability: 'capability-id-1',
+				command: 'command-1',
 				arguments: [],
-			}],
+			}
+			const expectedCommand2 = {
+				component: 'component-id',
+				capability: 'capability-id-2',
+				command: 'command-2',
+				arguments: ['arg1', 'arg2'],
+			}
+
+			expect(postSpy).toHaveBeenCalledTimes(1)
+			expect(postSpy).toHaveBeenCalledWith('device-id/commands', { commands: [expectedCommand1, expectedCommand2] })
+		})
+
+		it('throws exception if config entry is missing device config', async () => {
+			const configEntry = { valueType: ConfigValueType.DEVICE }
+			await expect(devices.sendCommand(configEntry, 'capability-id', 'command', ['arg']))
+				.rejects.toThrow('Device config not found')
+
+			expect(postSpy).toHaveBeenCalledTimes(0)
 		})
-		expect(axios.request).toHaveBeenCalledWith(expectedRequest(turnOn1.request))
-		expect(response).toBe(SuccessStatusValue)
 	})
 
-	it('sendCommands', async () => {
-		axios.request.mockImplementationOnce(() => Promise.resolve({ status: 200, data: turnOn1.response }))
-		const configEntry = {
-			'valueType': ConfigValueType.DEVICE,
-			'deviceConfig': {
-				'deviceId': '385931b6-0121-4848-bcc8-54cb76436de1',
-				'componentId': 'main',
-				'permissions': [
-					'r:devices:385931b6-0121-4848-bcc8-54cb76436de1',
-					'x:devices:385931b6-0121-4848-bcc8-54cb76436de1',
-				],
-			},
-		}
-		const response: Status[] = await client.devices.sendCommands([configEntry], 'switch', 'on')
-		expect(axios.request).toHaveBeenCalledWith(expectedRequest(turnOn1.request))
-		expect(response.length).toEqual(1)
-		expect(response[0]).toBe(SuccessStatusValue)
+	describe('sendCommands', () => {
+		const sendCommandSpy = jest.spyOn(devices, 'sendCommand')
+		const configEntry1 = { id: 'config-entry-1' }
+		const configEntry2 = { id: 'config-entry-2' }
+
+		it('does nothing given no config entries', async () => {
+			expect(await devices.sendCommands(undefined as unknown as [], 'capability-id', 'command')).toEqual([])
+
+			expect(sendCommandSpy).toHaveBeenCalledTimes(0)
+		})
+
+		it('uses sendCommand to send commands', async () => {
+			const configEntries = [configEntry1] as unknown as ConfigEntry[]
+			sendCommandSpy.mockResolvedValue(commandResponse)
+
+			expect(await devices.sendCommands(configEntries, 'capability-id', 'command', ['arg']))
+				.toEqual([commandResponse])
+
+			expect(sendCommandSpy).toHaveBeenCalledTimes(1)
+			expect(sendCommandSpy).toHaveBeenCalledWith(configEntry1, 'capability-id', 'command', ['arg'])
+		})
+
+		it('rejects on any failure', async () => {
+			// This reflects how the method currently works but it would be nice if the user could
+			// see how each individual command went.
```

Comment:
We should change `sendCommands` to use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) instead. This would allow the otherwise independent commands to succeed/fail without stopping if one fails. We could also introduce this as a "fail fast = false" flag to optionally send the commands this way in a future release (as to not be another breaking change).

Reply:
I did this now since it would be a breaking change and this PR is already a breaking change. (When I wrote this test I was still trying to keep it from being one.)

I had to update the typescript output libs to `es2020` to include `allSettled`. I also updated TypeScript while I was at it and ran `npm audit fix`.

I've included these new updates as a separate commit for easy review and I think I'll leave them that way since they are really unrelated to the other commit if that's okay with everyone else.

Reply:
I also ended up updating eslint. For some reason I couldn't run it on my machine.

Also, the new version of TypeScript requires a type on `catch` and it only allows `any` or `unknown`. I used any for now but we can revisit.